pkg/apis/v1beta1/nodepool.go

```diff
@@ -202,3 +210,72 @@ func (pl *NodePoolList) OrderByWeight() {
 		return ptr.Int32Value(pl.Items[a].Spec.Weight) > ptr.Int32Value(pl.Items[b].Spec.Weight)
 	})
 }
+
+// GetAllowedDisruptions returns the minimum allowed disruptions across all disruption budgets for a given node pool.
+// This returns two values as the resolved value for a percent depends on the number of current node claims.
+func (in *NodePool) GetAllowedDisruptions(ctx context.Context, c clock.Clock) (intstr.IntOrString, intstr.IntOrString, error) {
+	vals := make([]intstr.IntOrString, len(in.Spec.Disruption.Budgets))
+	errs := make([]error, len(in.Spec.Disruption.Budgets))
+	workqueue.ParallelizeUntil(ctx, len(in.Spec.Disruption.Budgets), len(in.Spec.Disruption.Budgets), func(i int) {
+		val, err := in.Spec.Disruption.Budgets[i].GetAllowedDisruptions(c)
+		if err != nil {
+			errs[i] = fmt.Errorf("invalid budget %s, %w", lo.FromPtr(in.Spec.Disruption.Budgets[i].Crontab), err)
+		}
+		vals[i] = val
+	})
+	if err := multierr.Combine(errs...); err != nil {
+		return intstr.IntOrString{}, intstr.IntOrString{}, err
+	}
+	minIntVal, minPercentVal := math.MaxInt64, math.MaxInt64
+	for i := range vals {
+		val := vals[i]
+		// The crontab wasn't active
+		if val.IntVal == -1 {
+			continue
+		}
+		// This returns the percent value if it's a string, and the raw value if it's an int.
+		temp, err := intstr.GetScaledValueFromIntOrPercent(lo.ToPtr(val), 100, false)
+		if err != nil {
+			// Should almost never happen since this is validated when the nodepool is applied
```

Comment:
Can we guarantee that this should never happen? I'd like to avoid having errors that "pretty much can't ever happen" since I question why we should be handling the error in the code at all

Reply:
If the validation mechanism isn't able to validate/default an object for some reason, a user could get themselves into config error. What are your thoughts on this?